### PR TITLE
[iOS] Change when we updated the XF INavigationPageController

### PR DESCRIPTION
**Pull request by rmarinho**
### Description of Change ###

Our problem is the way we were updating the XF navigation stack, when a page was removed (popped) from the NavigationController via a native gesture (tap back button, or swipe from side), we were relying that the override PopViewController will be called the correct amount of times, but that's not true. If the user presses the back button very fast, PopViewController will suppress to be called if animation is still happening, the code that was trying to fix this wasn't working as it should for consecutive back button calls, usually more then 5/6 it would fail, resulting in pages being still on the NavigationStack.
To be more sure about when the page is Popped we use the override DidMoveToParentViewController , this can tell us when a VC is detach from is parent NavigationController, when we detect we are being removed we need to update the NacigationPage stack, the only exception here is if the used called RemovePage, where we pop it manually first from the navigation stack so we ignore when this happens on our method. 
UITest didn't work because tapping back was to slow. 

Removed not used projects 

There were some style fixes also. 

### Bugs Fixed ###

https://bugzilla.xamarin.com/show_bug.cgi?id=39908
https://bugzilla.xamarin.com/show_bug.cgi?id=42341

### API Changes ###

None

### PR Checklist ###

- [ ] Has tests (if omitted, state reason in description)
- [x] Rebased on top of master at time of PR
- [x] Changes adhere to coding standard
- [x] Consolidate commits as makes sense

### CODE REVIEW

File: Xamarin.Forms.Core.iOS.UITests/BaseTestFixture.cs [hunk 1]
```diff
@@ -50,43 +50,33 @@ protected virtual void FixtureSetup ()
...
-				// if at first you dont succeed
-				RunningApp.Restart ();
-			}
-
-			// Wrap the app in ScreenshotConditional so it only takes screenshots if the SCREENSHOTS symbol is specified
-			App = new ScreenshotConditionalApp(RunningApp.App);
-
-			App.SetOrientationPortrait ();
-			ScreenBounds = App.RootViewRect ();
-			NavigateToGallery ();
```
Comment (StephaneDelcroix) [inline Xamarin.Forms.Core.iOS.UITests/BaseTestFixture.cs]: could you avoid style "fixes", it greatly reduce the SNR and makes the PR harder to review. thanks

File: Xamarin.Forms.Platform.iOS/Renderers/NavigationRenderer.cs [hunk 2]
```diff
@@ -799,6 +747,12 @@ public Page Child
 				}
 			}
 
+			public bool IgnorePageBeingRemoved
```
Comment (StephaneDelcroix) [inline Xamarin.Forms.Platform.iOS/Renderers/NavigationRenderer.cs]: this looks similar to `_ignorePopCall`

**Comment by StephaneDelcroix** [comment]
lot of noise, looks like a hack (the previous implementation was a hack too :) ). Need @jassmith to review this

File: Xamarin.Forms.Core.Android.UITests/App.cs [hunk 3]
```diff
@@ -1,22 +0,0 @@
-﻿using NUnit.Framework;
```
Comment (rmarinho) [inline Xamarin.Forms.Core.Android.UITests/App.cs]: this isn't important

File: Xamarin.Forms.Core.iOS.UITests/App.cs [hunk 4]
```diff
@@ -1,29 +0,0 @@
-﻿using System;
```
Comment (rmarinho) [inline Xamarin.Forms.Core.iOS.UITests/App.cs]: this is also not important

File: Xamarin.Forms.Core.iOS.UITests/BaseTestFixture.cs [hunk 5]
```diff
@@ -16,22 +16,22 @@ internal abstract class BaseTestFixture
 	{
```
Comment (rmarinho) [inline Xamarin.Forms.Core.iOS.UITests/BaseTestFixture.cs]: Not important only style changes

File: Xamarin.Forms.Core.iOS.UITests/BaseTestFixture.cs (hunk 1 above)
Comment (rmarinho) [inline Xamarin.Forms.Core.iOS.UITests/BaseTestFixture.cs]: this is a little more that just style fixes as i removed the projects and cleaned up the code.. But yeah could be a different PR

File: Xamarin.Forms.Platform.iOS/Renderers/NavigationRenderer.cs (hunk 2 above)
Comment (rmarinho) [inline Xamarin.Forms.Platform.iOS/Renderers/NavigationRenderer.cs]: it's a different class, this now belongs to the ParentViewController

**Comment by jassmith** [comment]
This looks good, removes an old and terrible hack. :+1:
